### PR TITLE
geth random port + temp dir for tests

### PR DESCRIPTION
**Pull request by chfast**
1. Random port for geth.
2. Bounded get IPC path length (by placing the socket in system temp dir directly).
3. Restore temp dir for tests.

### CODE REVIEW

**Review by badb — CHANGES_REQUESTED** [review]
Some blender benchmark & verification tests fail on this branch on windows. Give me some time to investigate this issue.

File: golem/testutils.py [hunk 1]
```diff
@@ -24,7 +22,7 @@ class TempDirFixture(unittest.TestCase):
     @classmethod
     def setUpClass(cls):
         logging.basicConfig(level=logging.DEBUG)
-        cls.__root_dir = get_local_datadir('tests')
+        cls.__root_dir = tempfile.mkdtemp(prefix='golem-tests-')
```
Comment (badb) [inline golem/testutils.py]:
```
if is_windows():
    import win32api
     cls.__root_dir = win32api.GetLongPathName(cls.__root_dir)
``` 
should solve the issue

**Comment by Elfoniok** [comment]
Wouldn't it be better to provide this option from config file?

**Review by paulperegud — APPROVED** [review]

**Comment by paulperegud** [comment]
works on Linux; haven't managed to test it on Windows

**Review by mfranciszkiewicz — APPROVED** [review]
`test_opt_node` hangs on Windows 10 x64, but merging with `develop` seems to do the trick.

**Pull request merged**